### PR TITLE
[Downgrade PHP 8.1] Add DowngradeReadonlyPropertyRector

### PR DESCRIPTION
**Pull request by villfa**
I had to run the tests with PHP 8.1 to avoid a syntax error: https://phpstan.org/r/2a115211-9002-4800-9999-261b2f7d36a7

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                php: ['8.0']
+                php: ['8.1']
```
Comment (samsonasik) [inline .github/workflows/tests.yaml]: please revert to see if it cause the error, it should be ok with 8.0 as we have other downgrade 8.1 that working ok

Comment (villfa) [reply .github/workflows/tests.yaml]: It's reverted. Now you can see the error.

FYI I run my PRs on my own repo before opening them here, and I had the same error.

The 3 other downgrade 8.1 rectors you mentionned are syntactically valid before PHP 8.1, even if they would crash at runtime, that's why we didn't see this error before.

Comment (samsonasik) [reply .github/workflows/tests.yaml]: /cc @ondrejmirtes probably related with RichParser?

Comment (villfa) [reply .github/workflows/tests.yaml]: By tuning PHPStan's configuration it now works with PHP 8.1.

File: config/phpstan/parser.neon [hunk 2]
```diff
@@ -1,3 +1,6 @@
+parameters:
+    phpVersion: 80100
```
Comment (samsonasik) [inline config/phpstan/parser.neon:2]: could you check this when used in scoped version running php 7.x? You may can clone rector/rector and check against this directory:

https://github.com/rectorphp/rector/blob/5e0eef55a50c696ddb922556061c09f2a1570b58/.github/workflows/e2e.yaml#L21

Comment (villfa) [reply config/phpstan/parser.neon:2]: Seems ok:
```
root@99f6d3b4e61d:/code/rector/e2e/define-constant# ../../bin/rector process --dry-run --ansi

 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 [OK] Rector is done!


root@99f6d3b4e61d:/code/rector/e2e/define-constant# php -v
PHP 7.4.25 (cli) (built: Oct 22 2021 17:23:33) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies

root@99f6d3b4e61d:/code/rector/e2e/define-constant# git diff
diff --git a/config/phpstan/parser.neon b/config/phpstan/parser.neon
index 58a2839..2504a37 100644
--- a/config/phpstan/parser.neon
+++ b/config/phpstan/parser.neon
@@ -1,3 +1,6 @@
+parameters:
+    phpVersion: 80100
+
 services:
     defaultAnalysisParser:
         factory: @cachedRectorParser
root@99f6d3b4e61d:/code/rector/e2e/define-constant#
```

Comment (samsonasik) [reply config/phpstan/parser.neon:2]: Thank you for verify

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**